### ISSUE TITLE
Crash when trying to open a midi score that musescore 4.1.1 was able to open

### ISSUE THREAD

**Issue by breakmit-0** [musescore/MuseScore#21020]
### Issue type

Crash or freeze

### Bug description

A midi file i downloaded causes musescore 4 to crash when trying to open it.

Regular midi editors are able to open it.

Musescore 3 is able to open it and save it to mscz format.

The file saved by musescore 3 also crashes musescore 4

[files.zip](https://github.com/musescore/MuseScore/files/13939200/files.zip)

### Steps to reproduce

Open crash_with_ms4.mid or crash_with_ms4.zip is MuseScore4

### MuseScore Version

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore version (64-bit): 4.2.0-233521124, revision: github-musescore-musescore-eb8d33c

### Regression

Yes, this used to work in **MuseScore 4.1.1**
### Operating system

Windows 11

### Additional context

build number : 233521124

musescore 3 version that worked : OS: Windows 10 (10.0), Arch.: x86_64, MuseScore version (32-bit): 3.6.2.548021803, revision: github-musescore-musescore-3224f34

logs when trying to open the midi file:

```
2024-01-15T14:05:35.870 | WARN  | main_thread     | Qt              | QQmlComponent: Created graphical object was not placed in the graphics scene.
2024-01-15T14:05:36.856 | INFO  | main_thread     | ActionsDispatcher::dispatch | try call action: file-open
2024-01-15T14:05:36.856 | INFO  | main_thread     | ProjectActionsController::openProject | Try open project: url = "" , displayNameOverride = "" 
```

when trying to open the musescore 3 file
```
2024-01-15T14:04:12.676 | WARN  | main_thread     | Qt              | QQmlComponent: Created graphical object was not placed in the graphics scene.
2024-01-15T14:04:13.341 | INFO  | main_thread     | ActionsDispatcher::dispatch | try call action: file-open
2024-01-15T14:04:13.341 | INFO  | main_thread     | ProjectActionsController::openProject | Try open project: url = "" , displayNameOverride = "" 
2024-01-15T14:04:36.883 | INFO  | main_thread     | ProjectMigrator::migrateEngravingProjectIfNeed | success migration
```

full logs in attachment

**Comment by Jojo-Schmitz** [comment]
My 3.7 crashes on these too

**Comment by Jojo-Schmitz** [comment]
It crashes in code added by #19402

**Comment by Jojo-Schmitz** [comment]
At https://github.com/musescore/MuseScore/pull/19402/files#diff-5a11d1bca6a45bf72941a466c258a05f289ade4705820c2f9e833a7adaca1552R1789

**Comment by DmitryArefiev** [comment]
> It crashes in code added by #19402

Yes, it crashes in 4.2 and doesn't occur in **4.1.1**

**Comment by DmitryArefiev** [comment]
Fixed in #21039
